hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -39,7 +39,7 @@
 import org.apache.hudi.common.model.HoodieFileFormat;
 import org.apache.hudi.common.model.HoodieFileGroup;
 import org.apache.hudi.common.model.HoodieIndexDefinition;
-import org.apache.hudi.common.model.HoodieIndexMetadata;
+import org.apache.hudi.common.model.HoodieIndexFunctionalOrSecondaryIndexMetadata;
```

Comment:
the new name does not actually read properly

Reply:
Hi @xushiyan Any suggestions for the class name?
How about `HoodieFunctionalOrSecondaryIndexMetadata` As the class is relevant only for  these two indexes.

Reply:
Closing the PR. Java Doc can be added to explain the functionalities of the class. Thanks